### ISSUE TITLE
Support files read as "JSON with Comments"

### ISSUE THREAD

**Issue by markmsmith** [ghmcadams/vscode-lintlens#11]
The `.eslintrc` and `.eslintrc.json` config formats allow adding javascript-style comments:   https://eslint.org/docs/user-guide/configuring#configuration-file-formats  
This is really helpful if you want to explain why you're disabling a rule.  
Unfortunately, if you tell VS Code to interpret the file as "JSON with Comments" so it doesn't complain about these (by clicking on `JSON` in the bottom of the editor and selecting it from the choice of language modes), your awesome plugin stops working.  I have this setting configured in my `settings.json` with the following:
```
"files.associations": {
    ".eslintrc.json": "jsonc"
  }
```

Looking at your code, it might be possible to fix the problem by adding another case to [https://github.com/ghmcadams/vscode-lintlens/blob/master/parsers/DocumentParser.js#L16](DocumentParser.js):
```javascript
else if (vscode.languages.match({ pattern: '**/.eslintrc.json', scheme: 'file', language: 'json' }, document) > 0 ||
vscode.languages.match({ pattern: '**/.eslintrc.json', scheme: 'file', language: 'jsonc' }, document) > 0) {
    return new JSONParser(document);
}
```

**Comment by ghmcadams** [comment]
Support for jsonc has been added in e1b7ecf7742ac0b1415b10e5b9de0ae273836dd9

**Comment by markmsmith** [comment]
Awesome, thank you!
